spec/parsers/webp_parser_spec.rb

```diff
@@ -0,0 +1,121 @@
+    expect(result.format).to eq(:webp)
+    expect(result.has_multiple_frames).to eq(false)
+    expect(result.has_transparency).to eq(false)
+    expect(result.height_px).to eq(181)
+    expect(result.intrinsics).to be_nil
+    expect(result.orientation).to be_nil
+    expect(result.width_px).to eq(65)
+  end
+
+  it 'successfully parses lossless WebP files' do
+    result = subject.call(File.open(fixtures_dir + 'WEBP/lossless.webp', 'rb'))
+    expect(result).not_to be_nil
+    expect(result.content_type).to eq('image/webp')
+    expect(result.format).to eq(:webp)
+    expect(result.has_multiple_frames).to eq(false)
+    expect(result.has_transparency).to eq(false)
+    expect(result.height_px).to eq(181)
+    expect(result.intrinsics).to be_nil
+    expect(result.orientation).to be_nil
+    expect(result.width_px).to eq(65)
+  end
+
+  it 'successfully parses lossless WebP files with an alpha channel' do
+    result = subject.call(File.open(fixtures_dir + 'WEBP/lossless-alpha.webp', 'rb'))
+    expect(result).not_to be_nil
+    expect(result.content_type).to eq('image/webp')
+    expect(result.format).to eq(:webp)
+    expect(result.has_multiple_frames).to eq(false)
+    expect(result.has_transparency).to eq(true)
+    expect(result.height_px).to eq(181)
+    expect(result.intrinsics).to be_nil
+    expect(result.orientation).to be_nil
+    expect(result.width_px).to eq(65)
+  end
+
+  it 'successfully parses extended WebP files' do
+    result = subject.call(File.open(fixtures_dir + 'WEBP/extended.webp', 'rb'))
+    expect(result).not_to be_nil
+    expect(result.content_type).to eq('image/webp')
+    expect(result.format).to eq(:webp)
+    expect(result.has_multiple_frames).to eq(false)
+    expect(result.has_transparency).to eq(false)
+    expect(result.height_px).to eq(181)
+    expect(result.intrinsics).to be_nil
+    expect(result.orientation).to be_nil
+    expect(result.width_px).to eq(65)
+  end
+
+  it 'successfully parses extended WebP files with an alpha channel' do
+    result = subject.call(File.open(fixtures_dir + 'WEBP/extended-alpha.webp', 'rb'))
+    expect(result).not_to be_nil
+    expect(result.content_type).to eq('image/webp')
+    expect(result.format).to eq(:webp)
+    expect(result.has_multiple_frames).to eq(false)
+    expect(result.has_transparency).to eq(true)
+    expect(result.height_px).to eq(181)
+    expect(result.intrinsics).to be_nil
+    expect(result.orientation).to be_nil
+    expect(result.width_px).to eq(65)
+  end
+
+  it 'successfully parses extended WebP files with Exif metadata' do
+    result = subject.call(File.open(fixtures_dir + 'WEBP/extended-exif.webp', 'rb'))
+    expect(result).not_to be_nil
+    expect(result.content_type).to eq('image/webp')
+    expect(result.format).to eq(:webp)
+    expect(result.has_multiple_frames).to eq(false)
+    expect(result.has_transparency).to eq(false)
+    expect(result.height_px).to eq(181)
+    expect(result.intrinsics).not_to be_nil
```

Comment:
You could do this in a more expressive way with:

```
expect(...).to match(exif: { image_length: '...', ... })
```

This would ensure a hash-like match on the intrinsics, which would merge your 4 expectations into one.

Reply:
Ooh that's cool! It seems like doing it this way would require matching every single Exif property though (there are 39 of them in this image 😅)

Reply:
Ah, in that case you could slip in a `a_hash_including()` and match like that 😊